### PR TITLE
Add tests and fix uncovered issues.

### LINKED ISSUE #24: Test all major functionality.
Rapcom needs tests to cover all expected functionality. This is not intended to be exhaustive, but should at least handle the happy paths.

### PR DESCRIPTION
**Pull request by dangle**
Fixes #24

### CODE REVIEW

File: rapcom/dispatcher.py [hunk 1]
```diff
@@ -54,10 +54,11 @@ def main():
     _enable_logging(args.get('--log-level'))
     try:
         if args.get('<command>') == 'help':
-            subcommand = next(iter(args['<args>']), None)
+            subcommand = next(iter(args.get('<args>')), None)
             return _help(subcommand)
         else:
-            argv = [args.get('<command>')] + args.get('<args>', sys.argv[1:])
+            default_args = sys.argv[2 if args.get('<command>') else 1:]
```
Comment (zancas) [inline rapcom/dispatcher.py]: Cute.

File: examples/explicit_example/setup.py [hunk 2]
```diff
@@ -7,7 +7,7 @@
     install_requires=['rapcom'],
     entry_points={
         'console_scripts': [
-            'say = rapcom.__main__:main'
+            'say = rapcom.dispatcher:main'
```
Comment (zancas) [inline examples/explicit_example/setup.py]: +1

File: rapcom/dispatcher.py [hunk 3]
```diff
@@ -54,10 +54,11 @@ def main():
     _enable_logging(args.get('--log-level'))
     try:
         if args.get('<command>') == 'help':
-            subcommand = next(iter(args['<args>']), None)
+            subcommand = next(iter(args.get('<args>')), None)
```
Comment (zancas) [inline rapcom/dispatcher.py]: +1

File: rapcom/dispatcher.py [hunk 4]
```diff
@@ -91,15 +92,17 @@ def _norm(k):
...
         return k.lower().replace('-', '_')
 
-    if isinstance(func, types.FunctionType) or not hasattr(func, '__call__'):
-        params = _get_signature(func)
-    else:
-        params = _get_signature(func.__call__)[1:]
+    assert hasattr(func, '__call__'), (
+        'Cannot normalize parameters of a non-callable.')
+    is_func = isinstance(func, types.FunctionType)
+    params = _get_signature(func if is_func else func.__call__)
```
Comment (zancas) [inline rapcom/dispatcher.py]: TERRRRRRNARY...     yer a thoughtful coder.

File: rapcom/dispatcher.py [hunk 5]
```diff
@@ -357,6 +363,9 @@ def _enable_logging(log_level):
     """
     if log_level not in (None, 'DEBUG', 'INFO', 'WARN', 'ERROR'):
         raise ValueError('Invalid log level "{}".'.format(log_level))
+    if '--log-level' in sys.argv:
+        sys.argv.remove('--log-level')
+        sys.argv.remove(log_level)
```
Comment (zancas) [inline rapcom/dispatcher.py]: Uhhh....    what does line 368 do?

File: setup.py [hunk 6]
```diff
@@ -29,7 +29,7 @@
         'backports.shutil_get_terminal_size'
     ],
     setup_requires=['pytest-runner'],
-    tests_require=['pytest >= 2.9, < 3'],
+    tests_require=['pytest >= 2.9'],
```
Comment (zancas) [inline setup.py]: Shall we try `>=3.0.5, <4`?

File: tox.ini [hunk 7]
```diff
@@ -9,7 +9,7 @@ envlist =
 usedevelop = True
 passenv = TRAVIS TRAVIS_JOB_ID TRAVIS_BRANCH
 deps =
-    pytest >= 2.9, < 3
+    pytest >= 2.9
```
Comment (zancas) [inline tox.ini]: The default pytest for some systems is `3.0.5`....

File: rapcom/dispatcher.py (hunk 5 above)
Comment (dangle) [inline rapcom/dispatcher.py]: You pass in values like `--log-level DEBUG`. 367 removes `--log-level`, 368 removes `DEBUG`.

File: setup.py (hunk 6 above)
Comment (dangle) [inline setup.py]: Well, things didn't appear to break between 2.9 and 3.0, so I think it's best to stick with the lowest working value that we know of.

File: tox.ini (hunk 7 above)
Comment (dangle) [inline tox.ini]: This will cover that. :-)

**Pull request merged**